graylog2-server/src/main/java/org/graylog2/security/ldap/LdapSettingsImpl.java

```diff
@@ -140,8 +142,18 @@ public String getSystemPassword() {
                 getSystemPasswordSalt());
     }
 
+    @Override
+    public String getSystemPasswordPlaceholder() {
+        final Object o = fields.get(SYSTEM_PASSWORD);
+        if (o == null) return "";
+        return SYSTEM_PASSWORD_PLACEHOLDER;
+    }
+
     @Override
     public void setSystemPassword(String systemPassword) {
+        if (systemPassword == SYSTEM_PASSWORD_PLACEHOLDER) {
```

Comment:
The correct way to check for equality in java is using `Object#equals`. This way (using `==`) we check for reference equality, not value equality, which works only for primitive types or by accident (when equal values are managed as singletons).